src/sentry/api/endpoints/release_thresholds/release_threshold_status_index.py

```diff
@@ -126,61 +161,161 @@ def get(self, request: Request, organization: Organization | RpcOrganization) ->
+                    thresholds_list: List[ReleaseThreshold] = project.release_thresholds.filter(
                         environment__name__in=environments_list
                     )
                 else:
                     thresholds_list = project.release_thresholds.all()
                 for threshold in thresholds_list:
-                    is_healthy = self.is_threshold_healthy(threshold)
-                    project_threshold_statuses.append(
-                        {
-                            **serialize(threshold),
-                            "is_healthy": is_healthy,
+                    if threshold.threshold_type not in thresholds_by_type:
+                        thresholds_by_type[threshold.threshold_type] = {
+                            "projects": [],
+                            "releases": [],
+                            "thresholds": [],
                         }
+                    thresholds_by_type[threshold.threshold_type]["projects"].append(project.id)
+                    thresholds_by_type[threshold.threshold_type]["releases"].append(release.version)
+                    enriched_threshold: EnrichedThreshold = {
+                        # see https://github.com/python/mypy/issues/4122
+                        **serialize(threshold),  # type: ignore
+                        "key": self.construct_threshold_key(
+                            release=release, project=project, threshold=threshold
+                        ),
+                        "start": release.date,  # deploy.date_finished _would_ be more accurate, but is not keyed on project so cannot be used
+                        "end": release.date
+                        + timedelta(threshold.window_in_seconds),  # start + threshold.window
+                        "release": release.version,
+                        "project_id": project.id,
+                        "is_healthy": False,
+                    }
+                    thresholds_by_type[threshold.threshold_type]["thresholds"].append(
+                        enriched_threshold
                     )
-                release_project[project.id] = project_threshold_statuses
-            release_threshold_health[release.id] = release_project
+
+        # ========================================================================
+        # Step 4: Determine threshold status per threshold type and return results
+        # ========================================================================
+        release_threshold_health = defaultdict(list)
+        for threshold_type, filter_list in thresholds_by_type.items():
+            project_id_list = [proj_id for proj_id in filter_list["projects"]]
+            release_value_list = [release_version for release_version in filter_list["releases"]]
+            category_thresholds: List[EnrichedThreshold] = filter_list["thresholds"]
+            if threshold_type == ReleaseThresholdType.TOTAL_ERROR_COUNT:
```

Comment:
nit: match case makes this much more readable

Reply:
Match case is not available in the version of python we're running

Reply:
Ah also fwiw: https://www.youtube.com/watch?v=Bk-Gn7J6sz4

we should probably avoid match anyways